### ISSUE TITLE
Changing aws_network_interface param private_ips_count has no effect

### ISSUE THREAD

**Issue by Dogers** [terraform-providers/terraform-provider-aws#3210]
### Terraform Version
```
Terraform v0.11.2
+ provider.aws v1.8.0
```
### Affected Resource(s)
aws_network_interface

### Terraform Configuration Files
```hcl
resource "aws_network_interface" "privnic" {
  private_ips_count = 3
  ...
```

### Expected Behavior
Once the interface has been created, adding/increasing the private_ips_count parameter should cause the provider to allocate additional IP addresses to that ENI.

### Actual Behavior
When trying to use the expected new IP (as ${aws_network_interface.privnic.private_ips[2]}) Terraform throws an error:
```
Error: Error asking for user input: 1 error(s) occurred:

* local.nsvars: local.nsvars: index 2 out of range for list aws_network_interface.privnic.private_ips (max 2) in:

${aws_network_interface.privnic.private_ips[2]}
```

### Steps to Reproduce
The interface was originally created with private_ips_count = 2. This has now been increased to 3 and the new one cannot be used, as per the actual behavior section.

**Comment by Dogers** [comment]
Interestingly after a little bit of playing it seems that reducing the number of private IPs works, so this may just be a validation order issue?

**Comment by tigerpaw** [comment]
I ran into this recently, here's what I did to work around it: 

Assuming you have never set the `private_ips_count` parameter on your resource and you have only 1 private IP currently assigned to it. Add the parameter and value as `private_ips_count = 1`, run an apply then bump the value up to the number of private IPs you desire. 

It seems that if you have never set this parameter, it will assume its value is the number you set it to when you first set it.
